core/commands/destroy_command.rb

```diff
@@ -36,6 +36,7 @@ def check_parameters
   # @return [Configuration, String] parsed configuration.
   def setup_command
     @aws_service = @env.aws_service
+    return [nil, nil] if $session.node_name || $session.list
```

Comment:
This looks like a hack. If the old approach forces to makes hacks, then you must switch to the new one.

Reply:
This forces the use of `.nil?` method in several occasions. The variables by default must not have nil values.